addons/website_slides/static/tests/tours/slides_tour_tools.js

```diff
@@ -87,6 +95,126 @@ var addArticleToSection = function (sectionName, pageName, backend) {
 }];
 };
 
+const fillInFileInput = async (input, name, type, content) => {
+    const blob = await fetch(`data:${type};base64,${content}`).then(res => res.blob());
+    const file = new File([blob], name, { type: type});
+    const dataTransfer = new DataTransfer();
+    dataTransfer.items.add(file);
+    input.files = dataTransfer.files;
+    input.dispatchEvent(new Event('change', { bubbles: true }));
+};
+
+const compareBase64Content = async (url, name, type, expectedContent) => {
+    const blob = await fetch(url).then(res => res.blob());
+    const file = new File([blob], name, { type: type});
+    const actualContent = await utils.getDataURLFromFile(file).then(dataURL => dataURL.split(',', 2)[1]);
+    return expectedContent === actualContent;
+};
+
+/**
+ * Test the upload of an image file as a new content and especially the binary content of the uploaded image.
+ */
+const addImageToSection = (sectionName, pageName, backend) => {
+    return [
+{
+    content: 'eLearning: add content to section',
+    trigger: (backend ? 'iframe ' : '') + 'div.o_wslides_slide_list_category_header:contains("' + sectionName + '") a:contains("Add Content")',
+}, {
+    content: 'eLearning: click on image',
+    trigger: (backend ? 'iframe ' : '') + 'a[data-slide-category=infographic]',
+}, {
+    content: 'eLearning: choose upload from device',
+    trigger: (backend ? 'iframe ' : '') + '#source_type_local_file',
+}, {
+    content: 'eLearning: load image',
+    trigger: 'body',
+    run: async () => {
+        const uploadInput = backend ?
+            document.getElementsByTagName('iframe')[0].contentWindow.document.getElementById('upload') :
+            document.getElementById('upload');
+        await fillInFileInput(uploadInput, 'Overview.png', 'image/png', testPngImage);
+    },
+}, {
+    content: 'eLearning: ensure that the preview is displayed which means that data is loaded and can be submitted',
+    trigger: (backend ? 'iframe ' : '') + '#slide-image[src*="data:"]',
+}, {
+    content: 'eLearning: create and publish slide',
+    trigger: (backend ? 'iframe ' : '') + 'footer.modal-footer button:contains("Publish")',
+}, {
+    content: 'eLearning: launch content',
+    trigger: (backend ? 'iframe ' : '') + 'a.o_wslides_js_slides_list_slide_link:contains("Overview")',
+}, {
+    content: 'eLearning: check uploaded image presence and perform comparison',
+    trigger: (backend ? 'iframe ' : '') + '.o_wslides_fs_player img',
+    run: async () => {
+        const img = backend ? $('iframe').contents().find('.o_wslides_fs_player img') : $('.o_wslides_fs_player img');
```

Comment:
can you use document.querySelector instead of jquery ?

Reply:
Changed to use document.querySelector as suggested